src/ui/redux/actions/content.js

```diff
@@ -334,12 +180,66 @@ export function doFetchClaimsByChannel(uri: string, page: number = 1, pageSize:
   };
 }
 
-export function doPlayUri(uri: string) {
-  return (dispatch: Dispatch) => {
+export function doPurchaseUriWrapper(uri: string, cost: number, saveFile: boolean) {
+  return (dispatch: Dispatch, getState: () => any) => {
+    function onSuccess(fileInfo) {
+      dispatch(doUpdateLoadStatus(uri, fileInfo.outpoint));
+    }
+
+    // Only pass the sucess callback if we are saving the file, otherwise we don't show the download percentage
+    const successCallBack = saveFile ? onSuccess : undefined;
+    dispatch(doPurchaseUri(uri, { costInfo: cost }, saveFile, successCallBack));
+  };
+}
+
+export function doPlayUri(uri: string, skipCostCheck: boolean = false, saveFileOverride: boolean = false) {
+  return (dispatch: Dispatch, getState: () => any) => {
+    // Set the active playing uri so we can avoid showing error notifications if a previously started download fails
     dispatch(doSetPlayingUri(uri));
-    // @if TARGET='app'
-    dispatch(doPurchaseUri(uri));
-    // @endif
+
+    const state = getState();
+    const fileInfo = makeSelectFileInfoForUri(uri)(state);
+    const uriIsStreamable = makeSelectUriIsStreamable(uri)(state);
+    const downloadingByOutpoint = selectDownloadingByOutpoint(state);
+    const alreadyDownloaded = fileInfo && (fileInfo.completed || (fileInfo.blobs_remaining === 0 && uriIsStreamable));
```

Comment:
I've seen a few fileInfo checks similar to this, consider if it makes sense to combine into a selector (it might not).